src/main/java/com/ait/lienzo/client/core/shape/TextBoundsWrap.java

```diff
@@ -0,0 +1,153 @@
+/*
+   Copyright (c) 2017 Ahome' Innovation Technologies. All rights reserved.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+ */
+
+package com.ait.lienzo.client.core.shape;
+
+import java.util.ArrayList;
+
+import com.ait.lienzo.client.core.Context2D;
+import com.ait.lienzo.client.core.types.BoundingBox;
+import com.ait.lienzo.shared.core.types.TextAlign;
+import com.ait.lienzo.shared.core.types.TextBaseLine;
+import com.ait.lienzo.shared.core.types.TextUnit;
+
+/**
+ * ITextWrapper implementation that wraps text when a line exceeds the width of the provided boundary.
+ */
+public class TextBoundsWrap extends TextNoWrap {
+
+    private BoundingBox wrapBoundaries;
+
+    public TextBoundsWrap(final Text text) {
+        this(text,
+             new BoundingBox());
+    }
+
+    public TextBoundsWrap(final Text text,
+                          final BoundingBox wrapBoundaries) {
+        super(text);
+        setWrapBoundaries(wrapBoundaries);
+    }
+
+    public TextBoundsWrap(final ITextWrapper.Supplier<String> textSupplier,
+                          final ITextWrapper.Supplier<Double> fontSizeSupplier,
+                          final ITextWrapper.Supplier<String> fontStyleSupplier,
+                          final ITextWrapper.Supplier<String> fontFamilySupplier,
+                          final ITextWrapper.Supplier<TextUnit> textUnitSupplier,
+                          final ITextWrapper.Supplier<TextBaseLine> textBaseLineSupplier,
+                          final ITextWrapper.Supplier<TextAlign> textAlignSupplier,
+                          final BoundingBox wrapBoundaries) {
+        super(textSupplier,
+              fontSizeSupplier,
+              fontStyleSupplier,
+              fontFamilySupplier,
+              textUnitSupplier,
+              textBaseLineSupplier,
+              textAlignSupplier);
+        this.wrapBoundaries = wrapBoundaries;
+    }
+
+    public BoundingBox getWrapBoundaries() {
+        return wrapBoundaries;
+    }
+
+    public TextBoundsWrap setWrapBoundaries(final BoundingBox boundaries) {
```

Comment:
@manstis This setter seems to be used just in one of constructors above, is this setter used on purpose? I thought you need it for mocking in tests but I have not found any occurrence, I am not saying this has to be removed, I am just curious.

Reply:
@jomarko Hi.. the setter is used by Stunner (when re-sizing graph nodes the boundaries are updated so that text wraps within the boundaries of the shape). See [here](https://github.com/Christopher-Chianelli/kie-wb-common/blob/e94a3f48abe9c8fe48e92ae88d29590cbd595f72/kie-wb-common-stunner/kie-wb-common-stunner-client/kie-wb-common-stunner-lienzo/src/main/java/org/kie/workbench/common/stunner/client/lienzo/shape/view/wires/ext/WiresShapeViewExt.java#L206) (in a PR that is not merged at the moment.. I will need to change some existing/pending Stunner code when these Lienzo PRs are merged and we move to the resultant new version in Stunner).